src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,127 +135,378 @@ internal abstract class AbstractAddParameterCodeFixProvider<
-                            // to fix is something different.  That means there was an error earlier
-                            // than this argument.  Which means we're looking at a non-viable 
-                            // constructor.  Skip this one.
-                            continue;
+                            if (argumentOpt != null && argumentToAdd != argumentOpt)
+                            {
+                                // We were trying to fix a specific argument, but the argument we want
+                                // to fix is something different.  That means there was an error earlier
+                                // than this argument.  Which means we're looking at a non-viable 
+                                // constructor or method.  Skip this one.
+                                continue;
+                            }
+
+                            methodsAndArgumentToAdd.Add(new ArgumentInsertPositionData<TArgumentSyntax>(
+                                method, argumentToAdd, arguments.IndexOf(argumentToAdd)));
                         }
+                    }
+                }
+            }
+
+            return methodsAndArgumentToAdd.ToImmutableAndFree();
+        }
+
+        private int NonParamsParameterCount(IMethodSymbol method)
+            => method.IsParams() ? method.Parameters.Length - 1 : method.Parameters.Length;
+
+        private void RegisterFixForMethodOverloads(
+            CodeFixContext context,
+            SeparatedSyntaxList<TArgumentSyntax> arguments,
+            ImmutableArray<ArgumentInsertPositionData<TArgumentSyntax>> methodsAndArgumentsToAdd)
+        {
+            var codeFixData = PrepareCreationOfCodeActions(context.Document, arguments, methodsAndArgumentsToAdd);
 
-                        constructorsAndArgumentToAdd.Add(
-                            (constructor, argumentToAdd, arguments.IndexOf(argumentToAdd)));
+            // To keep the list of offered fixes short we create one menu entry per overload only
+            // as long as there are two or less overloads present. If there are more overloads we
+            // create two menu entries. One entry for non-cascading fixes and one with cascading fixes.
+            var fixes = codeFixData.Length <= 2
+                ? NestByOverload()
+                : NestByCascading();
+
+            context.RegisterFixes(fixes, context.Diagnostics);
+            return;
+
+            ImmutableArray<CodeAction> NestByOverload()
+            {
+                var builder = ArrayBuilder<CodeAction>.GetInstance(codeFixData.Length);
+                foreach (var data in codeFixData)
+                {
+                    // We create the mandatory data.CreateChangedSolutionNonCascading fix first.
+                    var title = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0, data.Method, includeParameters: true);
+                    CodeAction codeAction = new MyCodeAction(
```

Comment:
@jinujoseph @kuhlenh this seems another design question. do we want non cascading fix? is scenario major one?